lib/wanda/catalog/selected_check.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Wanda.Catalog.SelectedCheck do
+  @moduledoc """
+  Represents a selected check used during a check execution.
+
+  It carries information about the check specification and its available customizations.
+  """
+
+  alias Wanda.Catalog.{Check, CustomizedValue}
+
+  defstruct [
+    :id,
+    :spec,
+    :customized,
+    :customizations
+  ]
+
+  @type t :: %__MODULE__{
+          id: String.t(),
+          spec: Check.t(),
```

Comment:
nitpick: what if we name this `check`? It embeds a `Check`. Anyway, after understanding the whole flow, it is not that confusing.